src/components/vuestic-components/va-button/VaButton.vue

```diff
@@ -53,86 +53,40 @@ import {
   getBoxShadowColor,
 } from '../../../services/color-functions'
 import { ColorThemeMixin } from '../../../services/ColorThemePlugin'
-import { ContextPluginMixin, getContextPropValue } from '../../context-test/context-provide/ContextPlugin'
+import { getContextablePropMixin } from '../../context-test/context-provide/ContextPlugin'
 import { RouterLinkMixin } from '../../vuestic-mixins/RouterLinkMixin'
 
 export default {
   name: 'VaButton',
   components: { VaIcon },
-  mixins: [ColorThemeMixin, RouterLinkMixin, ContextPluginMixin],
-  inject: {
-    va: {
-      default: () => ({}),
-    },
-  },
-  props: {
-    color: {
-      type: String,
-      default () {
-        return getContextPropValue(this, 'color', 'success')
-      },
-    },
-    tag: {
-      type: String,
-      default () {
-        return getContextPropValue(this, 'tag', 'button')
-      },
-    },
-    outline: {
-      type: Boolean,
-      default () {
-        return getContextPropValue(this, 'outline', false)
-      },
-    },
-    flat: {
-      type: Boolean,
-      default () {
-        return getContextPropValue(this, 'flat', false)
-      },
-    },
-    size: {
+  mixins: [
+    ColorThemeMixin,
+    RouterLinkMixin,
+
+    getContextablePropMixin('color', { type: String, default: 'success' }),
```

Comment:
I suggest using the mixin factory before initializing the component.
This may be to make component rendering easier.

For example:
```
const contextMixinButton = getContextablePropMixinFuctory([
{'color', { type: String, default: 'success' }},
{'tag', { type: String, default: 'button' }},
...
])()

const component = {
   mixins: [ contextMixinButton ]
}
```

The second benefit can be splitting component config with component construction.